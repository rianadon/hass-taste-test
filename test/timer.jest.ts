import HassTest, { multiply } from '../src/hass-test'
import PlaywrightIntegration, { Element } from '../src/integrations/playwright'
import { toMatchImageSnapshot } from 'jest-image-snapshot'

const CONFIGURATION_YAML = `
timer:
  basic:
    duration: "00:01:00"
  laundry:
    duration: "00:01:00"
`

let hass: HassTest<Element>;

expect.extend({ toMatchImageSnapshot });

beforeAll(async () => {
    hass = new HassTest(CONFIGURATION_YAML, {
        integration: new PlaywrightIntegration(process.env.BROWSER || 'firefox')
    })
    await hass.start()
}, 10000)

afterAll(async () => await hass.close())

it('Timer element', async () => {
    const dashboard = await hass.Dashboard([
        { type: 'entities', entities: ['timer.basic'] }
    ])
    expect(await dashboard.cards[0].narrow('hui-timer-entity-row').html()).toMatchSnapshot();
    expect(await dashboard.cards[0].narrow('hui-timer-entity-row').screenshot()).toMatchImageSnapshot();
})

it('Timer states', async () => {
    const dashboard = await hass.Dashboard([
        { type: 'entities', entities: ['timer.laundry'] }
    ])
    // This is merely a reference to the element, so it can be reused
    const entityRow = dashboard.cards[0].narrow('.text-content')

    expect(await entityRow.text()).toMatchInlineSnapshot(`"Idle"`)

    await hass.callService('timer', 'start', {}, { entity_id: 'timer.laundry' })
    expect(await entityRow.text()).toMatchInlineSnapshot(`"59"`)

    await hass.callService('timer', 'pause', {}, { entity_id: 'timer.laundry' })
    expect(await entityRow.text()).toMatchInlineSnapshot(`"1:00 (Paused)"`)

    await hass.callService('timer', 'cancel', {}, { entity_id: 'timer.laundry' })
    expect(await entityRow.text()).toMatchInlineSnapshot(`"Idle"`)
})
