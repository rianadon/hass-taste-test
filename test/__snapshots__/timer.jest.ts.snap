// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timer element 1`] = `
"<hui-timer-entity-row>
  <hui-generic-entity-row config="[object Object]" hideName="false" no-secondary="">
    <state-badge _showIcon="true" class="pointer" stateObj="[object Object]" tabindex="0">
      <ha-state-icon data-domain="timer" data-state="idle" style="">â€¦</ha-state-icon>
    </state-badge>
    <div class="info pointer text-content" title="basic">
      basic
    </div>
    <div class="pointer text-content value">
      <div class="state">
        <slot></slot>
      </div>
    </div>
    <div class="text-content">
      Idle
    </div>
  </hui-generic-entity-row>
</hui-timer-entity-row>
"
`;
